src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/AnalyzerExtensions.cs

```diff
@@ -4,43 +4,20 @@
 using System;
 using System.Threading;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Operations;
 
-namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default
 {
-    internal static class DiagnosticAnalyzerExtensions
+    public static class AnalyzerExtensions
     {
-        public static NameSyntax GetFullName(this NameSyntax nameSyntax)
-        {
-            while (nameSyntax.Parent is QualifiedNameSyntax qualifiedParent)
-            {
-                nameSyntax = qualifiedParent;
-            }
-
-            return nameSyntax;
-        }
-
-        public static bool NameEquals(this IAssemblySymbol? symbol, string name, bool startsWith = true)
+        public static void RegisterMemberAccess(this AnalysisContext context, Action<InvocationAnalysisContext> action)
```

Comment:
Does the `SystemDeploymentAnalyzer` really need to register for all of these triggers?

Reply:
When should I reuse this method?

Reply:
Good question. It does seem like the symbol action would be sufficient. I'm just moving this code, though, so I'm leaving it as is for now.